### ISSUE TITLE
Passwd is mandatary in zabbix_user

### ISSUE THREAD

**Issue by exc1te** [ansible-collections/community.zabbix#232]
##### SUMMARY
Hello, when I'm using zabbix_user module to create users it says that 'passwd' is mandatory field, but when you want to create user that will be authenticated via LDAP this field can be omitted.

> Can be omitted if user is added only to groups that have LDAP access

[https://www.zabbix.com/documentation/current/manual/api/reference/user/create](url)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
zabbix_user

##### ANSIBLE VERSION
```
  ansible 2.9.10
  config file = /home/dskryminskiy/ansible/ansible.cfg
  configured module search path = [u'/home/dskryminskiy/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/dskryminskiy/virtualenv/ansible2910-python27/lib/python2.7/site-packages/ansible
  executable location = /home/dskryminskiy/virtualenv/ansible2910-python27/bin/ansible
  python version = 2.7.5 (default, Apr  2 2020, 13:16:51) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

##### OS / ENVIRONMENT / Zabbix Version
Zabbix 5.0.3

##### EXPECTED RESULTS
User that will be authenticated via LDAP do not need 'passwd' field.

##### ACTUAL RESULTS
User cannot be added without 'passwd' field.

**Comment by D3DeFi** [comment]
Thanks for reporting this @exc1te . I wonder if it will work as advertised, usually when API documentation states that something is required, then it will fail whole HTTP POST if not provided as a parameter to call.

If I understand it correctly, `hostgroup` should have `gui_access=2`, then it tells Zabbix that it uses LDAP. I have no easy way to test this with working LDAP setup, but I will try to verify this with just setting above mentioned parameter on a group later.

**Comment by D3DeFi** [comment]
@exc1te would you be able to test my PR #240 that addresses this issue please?

**Comment by exc1te** [comment]
Hi @D3DeFi I made a two checks:
1. Add user in group with LDAP auth without password and authenticate then via LDAP - succeeded.
2. Add user in group with Internal auth without password and it fails:
`
The full traceback is:
Traceback (most recent call last):
  File "<stdin>", line 102, in <module>
  File "<stdin>", line 94, in _ansiballz_main
  File "<stdin>", line 40, in invoke_module
  File "/usr/lib64/python2.7/runpy.py", line 176, in run_module
    fname, loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib64/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/tmp/ansible_community.zabbix.zabbix_user_payload_29YFCG/ansible_community.zabbix.zabbix_user_payload.zip/ansible_collections/community/zabbix/plugins/modules/zabbix_user.py", line 638, in <module>
  File "/tmp/ansible_community.zabbix.zabbix_user_payload_29YFCG/ansible_community.zabbix.zabbix_user_payload.zip/ansible_collections/community/zabbix/plugins/modules/zabbix_user.py", line 597, in main
TypeError: fail_json() takes exactly 1 argument (2 given)
failed: [test] (item={u'usrgrps': [u'Local'], u'alias': u'test', u'surname': u'test surname', u'name': u'test name', u'type': u'Zabbix super admin'}) => {
    "ansible_loop_var": "item", 
    "changed": false, 
    "item": {
        "alias": "test", 
        "name": "test name", 
        "surname": "test surname", 
        "type": "Zabbix super admin", 
        "usrgrps": [
            "Local"
        ]
    }, 
    "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib64/python2.7/runpy.py\", line 176, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_community.zabbix.zabbix_user_payload_29YFCG/ansible_community.zabbix.zabbix_user_payload.zip/ansible_collections/community/zabbix/plugins/modules/zabbix_user.py\", line 638, in <module>\n  File \"/tmp/ansible_community.zabbix.zabbix_user_payload_29YFCG/ansible_community.zabbix.zabbix_user_payload.zip/ansible_collections/community/zabbix/plugins/modules/zabbix_user.py\", line 597, in main\nTypeError: fail_json() takes exactly 1 argument (2 given)\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", 
    "rc": 1
}
`

**Comment by D3DeFi** [comment]
@exc1te thanks for catching that. I did a stupid syntax error:
https://github.com/ansible-collections/community.zabbix/pull/240/commits/0b1e71792a7f5c6513f6e23deedf540df1bc973b

Should work now.

**Comment by exc1te** [comment]
@D3DeFi I made additional checks:
1. Add user in group with Internal auth without password - succeeded (User password is required. One or more groups are not LDAP based)
2. Add user un two groups (with LDAP and Internal auth) without password - succeeded (User password is required. One or more groups are not LDAP based)
Now all clear, thanks for the fix.

**Comment by D3DeFi** [comment]
Thank you for checking & testing. Will be included in the `1.1.0` release hopefully sometimes next week :)
